src/ovec.rs

```diff
@@ -177,7 +177,9 @@ where
             .map_err(|e| Error::Memory(e.to_string()))?;
 
         let second_batch = {
-            let cur_header = first_batch[0]
+            let cur_header = first_batch
+                .first()
+                .ok_or_else(|| Error::MissingValue(self.a.clone(), 0))?
```

Comment:
This is a hard error. In theory it should never happen when a memory that correctly implements the `MemoryADT` trait is used. I would therefore use a more explicit error message.

Reply:
No, just raise a `Error::Memory` with a clear message, mayb something like "no result found after calling `bread`, this should not happen and is likely due to a bug in the underlying memory implementation"

Reply:
done in 42a347ea9cebd6392499c8933cc7de869946bcea